iot-hub/Samples/service/RoleBasedAuthenticationSample/Program.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Azure.Identity;
+using CommandLine;
+using Microsoft.Azure.Devices;
+using System;
+using System.Threading.Tasks;
+
+namespace RoleBasedAuthenticationSample
+{
+    public class Program
+    {
+        /// <summary>
+        /// A sample to illustrate how to use Azure active directory for authentication to the IoT hub.
+        /// <param name="args">Run with `--help` to see a list of required and optional parameters.</param>
+        public static async Task Main(string[] args)
+        {
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            // Initialize Azure active directory credentials.
+            Console.WriteLine("Creating token credential.");
+
```

Comment:
This was the recommended way in track 2 to create Azure.Identity credential in SDK samples so following that pattern.

Reply:
I'm curious if Java has a similar pattern that it should follow. Where did you get this recommendation for .NET from?

Reply:
One can only specify the values via environment variables, so given we take them as parameters, this is the only way for us to specify them.

Reply:
I got this recommendation from the SDK team when we were writing this for ADT. It might be there for Java as well. Would be good to check. The auth folks want us to use DefaultAzureCredential. And if users wants other things, they want them to refer to Azure.Identiy.